src/matlab2tikz.m

```diff
@@ -1000,12 +1032,97 @@ case guitypes()
 end
 end
 % ==============================================================================
+function m2t = getPlotyyReferences(m2t,axisHandle)
+% Retrieve references to legend entries of the main plotyy axis
+
+% Retrieve legend handle
+if isAxisMain(axisHandle)
+    legendHandle = m2t.axesContainers{end}.LegendHandle;
+elseif isAxisPlotyy(axisHandle)
+    legendHandle = getAssociatedLegend(m2t,getPlotyyPeer(axisHandle));
+    m2t.axesContainers{end}.LegendHandle = legendHandle;
+end
+
+% No legend
+if ~isAxisPlotyy(axisHandle) || isempty(legendHandle)
+    m2t.axesContainers{end}.PlotyyReferences = [];
+
+elseif isAxisMain(axisHandle)
+    % Create plotyy references to legend entries of the main axis 
+    legendEntries = m2t.axesContainers{end}.LegendEntries;
+    ancAxes       = ancestor(legendEntries,'axes');
+    idx           = ismember(double([ancAxes{:}]), axisHandle);
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(idx);
+    
+    % Do not create a legend on the main axis
+    m2t.axesContainers{end}.LegendHandle = [];
+else
+    % Get legend entries associated to secondary plotyy axis
+    legendEntries = getLegendEntries(m2t);
+    ancAxes       = ancestor(legendEntries,'axes');
+    if iscell(ancAxes)
+        ancAxes = [ancAxes{:}]; 
+    end
+    idx = ismember(double(ancAxes), axisHandle);
+    m2t.axesContainers{end}.LegendEntries = legendEntries(idx);
+    
+    % Reference the main axis legend entries
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(~idx);
+end
+end
+% ==============================================================================
+function bool = isAxisMain(h)
+% Check if it is the main axis e.g. in a plotyy plot
+
+if ~isAxisPlotyy(h)
+    bool = true;
+    return
+end
+
+% If it is a Plotyy axis
+switch getEnvironment()
+    case 'Octave'
+        plotyyAxes = get(h, '__plotyy_axes__');
+        bool       = find(plotyyAxes == h) == 1;
+                    
+    case 'MATLAB'
+        bool = ~isempty(getappdata(h, 'PlotYYListenerManager'));
+end
+end
+% ==============================================================================
+function bool = isAxisPlotyy(h)
+% Check if handle is a plotyy axis
+
+switch getEnvironment()
+    case 'Octave'
+        try 
+            get(h, '__plotyy_axes__');
```

Comment:
When you perform `get(h)`, is `__plotyy_axes__` a field name or is it a *hidden* property? If the first case, we could get rid of the `try ... catch` statement.